src/arduino.c

```diff
@@ -37,6 +37,45 @@
 #include "stk500.h"
 #include "arduino.h"
 
+static int arduino_parseextparms(const PROGRAMMER *pgm, const LISTID extparms) {
+  const char *extended_param;
+  int attempts;
+  int rv = 0;
+
+  for (LNODEID ln = lfirst(extparms); ln; ln = lnext(ln)) {
+    extended_param = ldata(ln);
+
+    if (sscanf(extended_param, "attempts=%2d", &attempts) == 1) {
+      PDATA(pgm)->retry_attempts = attempts;
+      pmsg_info("setting number of retry attempts to %d\n", attempts);
+      continue;
+    }
+
+    if(str_starts(extended_param, "no_autoreset")) {
+      if(!str_eq(extended_param, "no_autoreset")) {
+        pmsg_error("invalid -xno_autoreset value %s. Use -xno_autoreset\n", extended_param);
+        rv = -1;
+        break;
+      }
+      PDATA(pgm)->autoreset = false;
+      continue;
+    }
+
+    if (str_eq(extended_param, "help")) {
+      msg_error("%s -c %s extended options:\n", progname, pgmid);
+      msg_error("  -xattempts=<arg> Specify no. connection retry attempts\n");
+      msg_error("  -xno_autoreset   Don't toggle RTS/DTR lines on port open to prevent a hardware reset\n");
+      msg_error("  -xhelp           Show this help menu and exit\n");
+      return LIBAVRDUDE_EXIT;
+    }
+
+    pmsg_error("invalid extended parameter '%s'\n", extended_param);
```

Comment:
In the last messaging clean-up the quotation marks tended to be removed, `"invalid extended parameter %s\n"` is the more recent AVRDUDE style.